lib/galaxy/tools/actions/__init__.py

```diff
@@ -413,7 +413,7 @@ def handle_output(name, output, hidden=None):
             if not completed_job and trans.app.config.legacy_eager_objectstore_initialization:
                 # Must flush before setting object store id currently.
                 trans.sa_session.flush()
-                object_store_populator.set_object_store_id(data)
+                object_store_populator.set_object_store_id(data=data, user=trans.user)
```

Comment:
Seems like ObjectStorePopulator could be given the user when it is created and not need to changed down here. I think I would prefer that.

Reply:
Done.